getline.ts/src/blockchain.ts

```diff
@@ -0,0 +1,237 @@
+import * as Web3 from 'web3';
+import {BigNumber} from 'bignumber.js';
+import * as moment from 'moment';
+
+import {Address} from './common';
+import {MetabackendClient, pb} from './metabackend';
+
+/**
+ * A connector used to access blockchains.
+ */
+export interface Blockchain {
+    /**
+     * Convert a given block number in the future to a timestamp when it lands
+     * on the blockchain.
+     * @param current Current block number.
+     * @param target Block number in the future.
+     * @returns Timestamp of target block number.
+     */
+    blockToTime(current: BigNumber, target: BigNumber): moment.Moment;
+    /**
+     * Returns current top block number on the blockchain.
+     * @returns Block number.
+     */
+    currentBlock(): Promise<BigNumber>;
+    /**
+     * Retrieves a handle to a contract at an address on the blockchain.
+     * @param name Name of contract on the metabackend.
+     * @param address Address of contract on the blockchain.
+     * @returns Contract at this address.
+     */
+    existing(name: string, address: Address): Promise<Contract>;
```

Comment:
How about an statically typed enum for the names?

Reply:
Doesn't really make sense to ingrain it at type level in this method, as it by design knows nothing about what sort of contracts the metabackend contains - it's just a generic loader. I could add another indirection function that then has a typed signature, but I think that's overkill.